fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/cursors/aggregate/AggregateCursor.java

```diff
@@ -98,6 +101,25 @@ public CompletableFuture<RecordCursorResult<QueryResult>> onNext() {
             }
             // Use the last valid result for the continuation as we need non-terminal one here.
             RecordCursorContinuation continuation = Verify.verifyNotNull(previousValidResult).getContinuation();
+            /*
+            Here is how continuation(previousValidResult) moves:
```

Comment:
Okay, I had a bit of trouble understanding the main idea of this comment, but I think it could work with a few tweaks. First, it would be good to start out with something like, "Update the `previousValidResult` to the next continuation even though it hasn't been returned. This is to return the correct continuation when there are single-element groups." Then I think the table you have makes sense, but without that pre-amble, I glazed over the `previousValidResult = row2` line on 110, and on line 113, it's not clear why `row3` is necessarily of a different group.

On the other hand, I don't want to do too much bike shedding here, because I suspect that we'll change some of this when we re-do things to handle out of bounds limits.